src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -75,19 +76,23 @@ public DrsService(SnapshotService snapshotService,
     }
 
     public void decrement() {
-        currentDRSRequests--;
+        int currentValue = currentDRSRequests.get();
+        int newValue = currentValue - 1;
+        currentDRSRequests.set(newValue);
     }
 
     public void increment() {
-        currentDRSRequests++;
+        int currentValue = currentDRSRequests.get();
```

Comment:
Same, this should be a single operation (that's one of the reasons for using an atomic).  I'm also not sure that you need these methods given that these should be a single line

Reply:
Oh, I guess if they're being called from somewhere else.  These should probably be named something more meaningful and have a javadoc comment

Reply:
I think I still do, as I want them to be changing the state in the Service from thee Controller, right?